### PR TITLE
add isGesture to onRegionChangeComplete iOS

### PR DESCRIPTION
**Pull request by eduardorfreitas93**
Reopen this PR by adding isGesture to onRegionChangeComplete callback on iOS MapKit

### Does any other open PR do the same thing?

https://github.com/react-native-maps/react-native-maps/pull/4113
https://github.com/react-native-maps/react-native-maps/pull/4272

### What issue is this PR fixing?

https://github.com/react-native-maps/react-native-maps/issues/4112

### How did you test this PR?

Tested in production and 1 simulator iOS

### CODE REVIEW

**Comment by eduardorfreitas93** [comment]
[RyanTG](https://github.com/RyanTG) If they don't approve this PR you can try using
https://javascript.plainenglish.io/live-npm-package-patching-with-patch-package-b8f30d47779c
https://gist.github.com/eduardorfreitas93/eb1f37784a936b4bd618ffd209fb72db

**Comment by Simon-TechForm** [comment]
Hi @eduardorfreitas93 👋. Thank you for opening a new PR after the refactor.

As far as I can tell, this PR adds`isGesture` to the `onRegionChange` with a value that's always false? That seems undesirable :) If you could look into that, I will be happy to review this again :)

**Comment by eduardorfreitas93** [comment]
@Simon-TechForm Hi, it adds isGesture true in the return of onRegionChangeComplete, but I can see if I implement it in onRegionChange too

**Comment by Simon-TechForm** [comment]
It's not a necessity to implement the `isGesture` functionality for `onRegionChange` for me to merge this. But we can't have it return `isGesture=false` even when it should be true :)

**Comment by eduardorfreitas93** [comment]
With this implementation I return isGesture true when the user makes the gesture to navigate the map, when the navigation on the map is automatic, for example zoom or recenter it has to return false, at least that was the functionality I needed

File: ios/AirMaps/AIRMapManager.m [hunk 1]
```diff
@@ -964,8 +964,10 @@ - (void)mapView:(AIRMap *)mapView regionDidChangeAnimated:(__unused BOOL)animate
 
     // Don't send region did change events until map has
     // started rendering, as these won't represent the final location
-    if (mapView.hasStartedRendering) {
-        [self _emitRegionChangeEvent:mapView continuous:NO];
+    if (mapView.hasStartedRendering && animated) {
+        [self _emitRegionChangeEvent:mapView continuous:NO isGesture:NO];
+    } else if (mapView.hasStartedRendering && !animated) {
+        [self _emitRegionChangeEvent:mapView continuous:NO isGesture:YES];
```
Comment (Simon-TechForm) [inline ios/AirMaps/AIRMapManager.m:970]: Since `animated` is used, the `__unused` macro should be removed from it.

**Review by Simon-TechForm — CHANGES_REQUESTED** [review]

**Comment by eduardorfreitas93** [comment]
@Simon-TechForm I made the changes

**Comment by Simon-TechForm** [comment]
@eduardorfreitas93 I'm afraid this approach isn't going to work either. If you swipe the map (meaning the region will keep changing for a bit after you release your gesture), `isGesture` will change to false even though it should be true.

I'm afraid that I'm too occupied atm to help you out with a solution, but I would suggest putting together a set of test cases that you evaluate when coming up with a new approach, e.g.:

| Case isGesture                           | Expected onRegionChange             | Actual onRegionChange          | Expected onRegionChangeComplete | Actual onRegionChangeComplete | Result |
|------------------------------------------|-------------------------------------|--------------------------------|---------------------------------|-------------------------------|--------|
| animateToRegion                          | false                               |                                | false                           |                               |        |
| animateToRegion intercepted by gesture   | false until interception, then true |                                | true                            |                               |        |
| Drag map                                 | true                                |                                | true                            |                               | Good   |
| Swipe map                                | true, also when gesture is released | true while gesture, then false | true                            | false                         | Bad    |
| Swipe map intercepted by animateToRegion | True until interception, then false |                                | false                           |                               |        |

These are just off the top of my head, you could come up with more/others. Also make sure to check out Google Maps to match the behaviour.

**Comment by Simon-TechForm** [comment]
I'll close this for now and re-open when we have a working solution :)

**Comment by wmonecke** [comment]
I think this is urgently needed. Have you guys found a workaround?

**Comment by RyanTG** [comment]
I don’t have a workaround. I created a “Discussion” for this here: https://github.com/react-native-maps/react-native-maps/discussions/4440

But it’s not gaining traction.

**Comment by wmonecke** [comment]
@RyanTG Awesome, thank you!

**Comment by samih-sghier** [comment]
This would be so useful haven't found a workaround

**Pull request closed without merging**